client/src/pages/course/mentor/confirm.tsx

```diff
@@ -75,21 +78,65 @@ function Page(props: { session: Session; courseAlias?: string }) {
     }
   };
 
-  if (course == null || (mentorData == null && loading)) {
+  if (loading) {
     return null;
   }
 
+  if (course == null || (mentorData == null && loading)) {
```

Comment:
But if loading is true you are returning null, so it's condition never be working.

Reply:
fixed